packaging/packaging.go

```diff
@@ -0,0 +1,293 @@
+	Cluster_id string   `json:"cluster_id"`
+	Version    string   `json:"version"`
+	Date       string   `json:"date"`
+	Files      []string `json:"files"`
+}
+
+func BuildLocalCSVFileList(stagingDirectory string) []string {
+	var csvList []string
+	fileList, err := ioutil.ReadDir(stagingDirectory)
+	if err != nil {
+		fmt.Println("could not read dir")
+		// log.Error(err, "Could not read the directory")
+	}
+	for _, file := range fileList {
+		if strings.Contains(file.Name(), ".csv") {
+			csvList = append(csvList, stagingDirectory+"/"+file.Name())
+		}
+	}
+	return csvList
+}
+
+func NeedSplit(filepath string) bool {
+	var totalSize int64 = 0
+	maxBytes := defaultMaxSize * megaByte
+	fileList, err := ioutil.ReadDir(filepath)
+	if err != nil {
+		fmt.Println("could not read dir")
+	}
+	for _, file := range fileList {
+		info, err := os.Stat(filepath + "/" + file.Name())
+		if err != nil {
+			return false
+		}
+		fileSize := info.Size()
+		totalSize += fileSize
+		if fileSize >= maxBytes || totalSize >= maxBytes {
+			return true
+		}
+	}
+	return false
+}
+
+func RenderManifest(logger logr.Logger, archiveFiles []string, cost *costmgmtv1alpha1.CostManagement, filepath string) (string, string) {
+	log := logger.WithValues("costmanagement", "RenderManifest")
+	// setup the manifest
+	manifestUUID, _ := uuidv4.Generate()
+	manifestDate := metav1.Now()
+	var manifestFiles []string
+	for idx, _ := range archiveFiles {
+		upload_name := manifestUUID + "_openshift_usage_report." + strconv.Itoa(idx) + ".csv"
+		manifestFiles = append(manifestFiles, upload_name)
+	}
+	fileManifest := Manifest{
+		Uuid:       manifestUUID,
+		Cluster_id: cost.Status.ClusterID,
+		Version:    cost.Status.OperatorCommit,
+		Date:       manifestDate.UTC().Format("2006-01-02 15:04:05"),
+		Files:      manifestFiles,
+	}
+	manifestFileName := filepath + "/manifest.json"
+	// write the manifest file
+	file, _ := json.MarshalIndent(fileManifest, "", " ")
+	_ = ioutil.WriteFile(manifestFileName, file, 0644)
+	// return the manifest file/uuid
+	log.Info("Generated manifest file", "manifest", manifestFileName)
+	return manifestFileName, manifestUUID
+}
+
+func addFileToTarWriter(logger logr.Logger, uploadName, filePath string, tarWriter *tar.Writer) error {
+	log := logger.WithValues("costmanagement", "addFileToTarWriter")
+	log.Info("Adding file to tar.gz", "file", filePath)
+	file, err := os.Open(filePath)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	stat, err := file.Stat()
+	if err != nil {
+		return err
+	}
+
+	header := &tar.Header{
+		Name:    uploadName,
+		Size:    stat.Size(),
+		Mode:    int64(stat.Mode()),
+		ModTime: stat.ModTime(),
+	}
+
+	err = tarWriter.WriteHeader(header)
+	if err != nil {
+		return err
+	}
+
+	_, err = io.Copy(tarWriter, file)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func WriteTarball(logger logr.Logger, tarFileName, manifestFileName, manifestUUID string, archiveFiles []string) string {
+	if len(archiveFiles) > 0 {
```

Comment:
```suggestion
	if len(archiveFiles) <= 0 {
		return ""
	}
```
and then un-nest what is below here